### PR TITLE
refactor(*): unify instanceof usage to pattern matching style

### LINKED ISSUE #6064: Inconsistent usage of `instanceof` patterns
## Description
The codebase currently has mixed usage of instanceof patterns:

- The modern pattern matching style (s instanceof String str) 
- The traditional style (s instanceof String) which requires explicit casting.

This inconsistency can lead to reduced code readability and potential confusion among contributors.

<details>
  <summary> Existing Code 1</summary>
  image
</details>

<details>
  <summary> Existing Code 2</summary>
  image
</details>

### Environment

- Kestra Version: develop

### PR DESCRIPTION
**Pull request by ori0o0p**
### What changes are being made and why?
The usage of instanceof has been standardized across the codebase.
This PR replaces traditional instanceof checks with pattern matching.
It improves code readability, removes redundant casts, and ensures consistent coding style.

---

### How the changes have been QAed?
The changes have been QAed by running all existing unit tests to ensure no regressions.
Example scenario:

``` java
Object input = "example";
if (input instanceof String str) {
    System.out.println("Length: " + str.length());
}
```

This scenario was verified in cases where instanceof with pattern matching is used to ensure compatibility.

---

### Setup Instructions
No special setup is required.

closes #6064

### CODE REVIEW

**Comment by ori0o0p** [comment]
I cleaned up the code by removing the redundant parentheses.

[Diff](https://github.com/kestra-io/kestra/compare/108e1d76e0147968a5f9aa1e370f744c4891ad80..422463ae381fed8a37e5de99e1c21dab29113b49?diff=split)

File: core/src/main/java/io/kestra/core/queues/QueueService.java
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/queues/QueueService.java:1]: This is not equivalent, please revert this change

**Review by loicmathieu — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/kestra/core/queues/QueueService.java
Comment (ori0o0p) [reply core/src/main/java/io/kestra/core/queues/QueueService.java:1]: I have reverted the change now.

https://github.com/kestra-io/kestra/pull/6073/commits/7b87a59bd60337f9c4c1d6d5057336453724a0ba

**Review by loicmathieu — APPROVED** [review]
LGTM, thanks

**Comment by ori0o0p** [comment]
Hold on. I made a mistake.....

**Comment by ori0o0p** [comment]
I've got it fixed.
